lib/Corres_UL.thy

```diff
@@ -905,12 +905,29 @@ qed
 lemma corres_inst: "corres_underlying sr nf nf' r P P' f g \<Longrightarrow> corres_underlying sr nf nf' r P P' f g" .
 
 lemma corres_assert_opt_assume:
-  assumes "\<And>x. P' = Some x \<Longrightarrow> corres_underlying sr nf nf' r P Q f (g x)"
-  assumes "\<And>s. Q s \<Longrightarrow> P' \<noteq> None"
-  shows "corres_underlying sr nf nf' r P Q f (assert_opt P' >>= g)" using assms
+  assumes "\<And>x. opt = Some x \<Longrightarrow> corres_underlying sr nf nf' r P Q f (g x)"
+  assumes "\<And>s. Q s \<Longrightarrow> opt \<noteq> None"
+  shows "corres_underlying sr nf nf' r P Q f (assert_opt opt >>= g)" using assms
```

Comment:
I think same as below, while we're here a fixup to `[| ...; ...|] ==> ...` form, removal of shows/assms and the `split:` on the next line would be nice.